web/modules/custom/dbcdk_community/src/Form/ProfileEditForm.php

```diff
@@ -0,0 +1,269 @@
+<?php
+
+/**
+ * @file
+ * Contains \Drupal\dbcdk_community\Form\ProfileEditForm.
+ */
+
+namespace Drupal\dbcdk_community\Form;
+
+use Drupal\dbcdk_community\CommunityTraits;
+use DBCDK\CommunityServices\Api\ProfileApi;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+use Symfony\Component\HttpFoundation\RequestStack;
+use Drupal\Core\DependencyInjection\ContainerInjectionInterface;
+use Drupal\Core\Form\FormBase;
+use Drupal\Core\Form\FormStateInterface;
+use Drupal\Core\Url;
+
+/**
+ * Edit a Community Service Profile.
+ */
+class ProfileEditForm extends FormBase implements ContainerInjectionInterface {
+
+  use CommunityTraits;
+
+  /**
+   * The current request stack.
+   *
+   * @var RequestStack $requestStack
+   */
+  protected $requestStack;
+
+  /**
+   * The DBCDK Community Service Profile API.
+   *
+   * @var ProfileApi $profileApi
+   */
+  protected $profileApi;
+
+  /**
+   * The Community Service Profile object from the username context.
+   *
+   * @var Profile $profile
+   */
+  protected $profile;
+
+  /**
+   * The date format the Community Client expects.
+   *
+   * @var string $dateFormat
+   */
+  protected $dateFormat = 'Y-m-d';
+
+  /**
+   * Creates a Profile Edit Form instance.
+   *
+   * @param RequestStack $request_stack
+   *   The current request stack.
+   * @param ProfileApi $profile_api
+   *   The DBCDK Community Service Profile API.
+   */
+  public function __construct(RequestStack $request_stack, ProfileApi $profile_api) {
+    $this->profileApi = $profile_api;
+    $this->requestStack = $request_stack;
```

Comment:
The RequestStack only seems to be needed for object creation. How about changing the contructor signatur to `Profile $profile, ProfileApi $profile_api` and handling the rest in `create`?